core-spark-adapter-2.2/src/main/scala/za/co/absa/spline/coresparkadapterapi/WriteCommandParserImpl.scala

```diff
@@ -16,10 +16,37 @@
 
 package za.co.absa.spline.coresparkadapterapi
 
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.execution.command.CreateDataSourceTableAsSelectCommand
 import org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand
 
-class WriteCommandParserImpl extends WriteCommandParser[SaveIntoDataSourceCommand] {
-  override def asWriteCommand(operation: SaveIntoDataSourceCommand): WriteCommand = {
-    WriteCommand(operation.options.getOrElse("path", ""), operation.mode, operation.provider, operation.query)
+class WriteCommandParserFactoryImpl extends WriteCommandParserFactory {
+  override def getWriteParser(): WriteCommandParser[LogicalPlan] = new WriteCommandParserImpl()
+
+  override def getSaveAsTableParser(): WriteCommandParser[LogicalPlan] = new SaveAsTableCommandParserImpl()
+}
+
+class SaveAsTableCommandParserImpl extends WriteCommandParser[LogicalPlan] {
+  override def matches(operation: LogicalPlan): Boolean = operation.isInstanceOf[CreateDataSourceTableAsSelectCommand]
+
+  override def asWriteCommand(operation: LogicalPlan): AbstractWriteCommand = {
+    val op = operation.asInstanceOf[CreateDataSourceTableAsSelectCommand]
+
+    val identifier = op.table.storage.locationUri match {
+      case Some(location) => location.toURL.toString
+      case _ => "table:/" + op.table.identifier.identifier
```

Comment:
Why database name isn't used in the identifier?

Reply:
If `table:/` is a predefined prefix for any `SaveAsTablerCommand` then it should probably be a encapsulated into the command.

Reply:
database name is part of identifier (if database name is not default)